src/qibocal/protocols/qubit_spectroscopies/qubit_spectroscopy.py

```diff
@@ -69,91 +69,223 @@ class QubitSpectroscopyData(ResonatorSpectroscopyData):
+            # If we're batching, update the LO
+            if lo_offset != 0 and lo_channels[qubit] is not None:
+                f0 = platform.config(drive_channels[qubit]).frequency
+                update_dict[lo_channels[qubit]] = {"frequency": f0 + lo_offset}
+
+            batch_updates.append(update_dict)
+
+        # Execute this batch
+        results = platform.execute(
+            [sequence],
+            [sweepers],
+            updates=batch_updates,
+            **params.execution_parameters,
         )
 
-    # Create data structure for data acquisition.
+        # Collect results from this batch
+        for qubit in targets:
+            result = results[ro_pulses[qubit].id]
+            f0 = platform.config(drive_channels[qubit]).frequency
+
+            signal = magnitude(result)
+            _phase = phase(result)
+
+            if len(signal.shape) > 1:
+                error_signal = np.std(signal, axis=0, ddof=1) / np.sqrt(signal.shape[0])
+                signal = np.mean(signal, axis=0)
+                error_phase = np.std(_phase, axis=0, ddof=1) / np.sqrt(_phase.shape[0])
+                _phase = np.mean(_phase, axis=0)
+            else:
+                error_signal = None
+                error_phase = None
+
+            # Store results with absolute frequencies
+            all_frequencies[qubit].append(delta_frequency_range + f0 + lo_offset)
+            all_signals[qubit].append(signal)
+            all_phases[qubit].append(_phase)
+            all_error_signals[qubit].append(error_signal)
+            all_error_phases[qubit].append(error_phase)
+
+    # Create data structure and aggregate results
     data = QubitSpectroscopyData(
         resonator_type=platform.resonator_type, amplitudes=amplitudes
     )
 
-    results = platform.execute(
-        [sequence],
-        [sweepers],
-        updates=[
-            {platform.qubits[q].probe: {"frequency": readout_frequency(q, platform)}}
-            for q in targets
-        ],
-        **params.execution_parameters,
-    )
+    # Combine all batches for each qubit
+    for qubit in targets:
+        # Concatenate arrays from all batches
+        freq = np.concatenate(all_frequencies[qubit])
+        signal = np.concatenate(all_signals[qubit])
+        _phase = np.concatenate(all_phases[qubit])
 
-    # retrieve the results for every qubit
-    for qubit, ro_pulse in ro_pulses.items():
-        result = results[ro_pulse.id]
-        # store the results
-        f0 = platform.config(platform.qubits[qubit].drive).frequency
-        signal = magnitude(result)
-        _phase = phase(result)
-        if len(signal.shape) > 1:
-            error_signal = np.std(signal, axis=0, ddof=1) / np.sqrt(signal.shape[0])
-            signal = np.mean(signal, axis=0)
-            error_phase = np.std(_phase, axis=0, ddof=1) / np.sqrt(_phase.shape[0])
-            _phase = np.mean(_phase, axis=0)
+        # Handle errors
+        if all_error_signals[qubit][0] is not None:
```

Comment:
Uhm, I assume that your intention here is to check whether one is not `None`, assuming the rest will be the same.

I'm not sure how much the assumption is reliable, or required. I don't see it likely to run a qubit spectroscopy in classification... (though it is certainly possible...)

I'm inclined to either remove the option at all, are make the correct check:
```suggestion
        if all(x is not None for x in all_error_signals[qubit]):
```
which is just more conservative.